src/Storages/System/StorageSystemColumns.cpp

```diff
@@ -140,32 +141,55 @@ class ColumnsSource : public ISource
 
             {
                 StoragePtr storage = storages.at(std::make_pair(database_name, table_name));
-                TableLockHolder table_lock = storage->tryLockForShare(query_id, lock_acquire_timeout);
+                TableLockHolder table_lock;
+                StorageMetadataPtr metadata_snapshot;
 
-                if (table_lock == nullptr)
+                /// Can throw UNKNOWN_DATABASE/UNKNOWN_TABLE in case of Alias table
+                try
                 {
-                    // Table was dropped while acquiring the lock, skipping table
-                    continue;
+                    table_lock = storage->tryLockForShare(query_id, lock_acquire_timeout);
```

Comment:
Let's also re-define `tryLockForShare` for `Alias`, and ignore exceptions inside it